docs/install-midstream.md

```diff
@@ -0,0 +1,151 @@
+# Installing and upgrading OpenShift Serverless midstream
+
+Navigate to resources directory:
+
+```shell
+cd resources-install-midstream
+```
+
+## Installation
+
+Create the `openshift-serverless` namespace:
+
+```shell
+oc apply -f 100-oss-namespace.yaml
+```
+
+Create an `OperatorGroup`:
+
+```shell
+oc apply -f 200-oss-operator-group.yaml
+```
+
+### Create a CatalogSource for each version
+
+For each branch of OSS you wish to install or upgrade, install a CatalogSource like the following:
+
+```yaml
+# CatalogSource for 1.24
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-v1-24-0
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-v1.24.0:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+---
+# CatalogSource for 1.25
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-v1-25-0
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-v1.25.0:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+---
+# CatalogSource for main
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  # Replace the version with whatever version you want to install
+  name: serverless-operator-nightly
+  namespace: openshift-marketplace
+spec:
+  displayName: Serverless Operator
+  # Replace the version with whatever version you want to install
+  image: registry.ci.openshift.org/knative/openshift-serverless-nightly:serverless-index
+  publisher: Red Hat
+  sourceType: grpc
+```
+
```

Comment:
Let's add a note here, that `main` is not recommended, since it likely more unstable than actual branches which usually back a released/supported version

Reply:
Done